src/presign/input.rs

```diff
@@ -49,26 +55,19 @@ impl Input {
             Err(CallerError::BadInput)?
         }
 
-        // There shouldn't be duplicates. Since we checked equality of the sets and the
-        // lengths already, this also applies to auxinfo (also, the `auxinfo::Output`
-        // type checks this at construction already).
+        // There shouldn't be duplicates.
+        // This check is redundant, since it's also checked in the `auxinfo::Output` and
+        // `keygen::Output` constructors, so we actually don't test it below.
+        // Also, since we checked equality of the sets and the lengths already, checking
+        // for keygen also validates it for auxinfo
         if key_pids.len() != keygen_output.public_key_shares().len() {
             error!("Duplicate participant IDs appeared in AuxInfo and KeyShare public input.");
             Err(CallerError::BadInput)?
         }
 
-        // The private key share should map to one of the public values.
-        // NB: The corresponding check for `auxinfo::Output` is handled in that type's
-        // constructor.
-        let expected_public_share = keygen_output.private_key_share().public_share()?;
-        if !keygen_output
-            .public_key_shares()
-            .iter()
-            .any(|public_share| expected_public_share == *public_share.as_ref())
-        {
-            error!("Keygen private share did not correspond to any of the provided keygen public shares.");
-            Err(CallerError::BadInput)?
-        }
+        // The constructors for keygen and auxinfo output already check other important
+        // properties, like that the private component maps to one of public
+        // components for each one.
```

Comment:
One check that I didn't include here that probably should be added is that the keygen and auxinfo private pieces correspond to the _same_ participant IDs.

Reply:
I added this check :)